packages/back-end/src/controllers/stripe.ts

```diff
@@ -178,12 +266,63 @@ export async function postWebhook(req: Request, res: Response) {
     }
 
     case "customer.subscription.deleted":
+    case "subscription_scheduled.canceled":
     case "customer.subscription.updated": {
+      console.log("updated webhook was hit");
+      console.log(event.data.object);
       const subscription = event.data
         .object as Stripe.Response<Stripe.Subscription>;
-      if (subscription) {
-        updateSubscription(subscription);
+
+      // Get the current subscription data instead of relying on a potentially outdated event
+      const currentStripeSubscriptionData = await stripe.subscriptions.retrieve(
+        subscription.id
+      );
+
+      // Get stripeCustomerId
+      const stripeCustomerId =
+        typeof currentStripeSubscriptionData.customer === "string"
+          ? currentStripeSubscriptionData.customer
+          : currentStripeSubscriptionData.customer.id;
+
+      // Get the organization connected to stripeCustomerId
+      const currentDbSubscription = await findOrganizationByStripeCustomerId(
+        stripeCustomerId
+      );
+
+      // If Stripe's status, trialEnd, or subscriptionId's don't match our DB, update our DB.
+      if (
+        currentStripeSubscriptionData.status !==
+          currentDbSubscription?.subscription?.status ||
+        currentStripeSubscriptionData.trial_end !==
+          currentDbSubscription?.subscription?.trialEnd ||
+        currentStripeSubscriptionData.id !==
+          currentDbSubscription.subscription.id
+      ) {
+        updateSubscription(currentStripeSubscriptionData);
       }
+
+      // // This is a bit weird, but the organization.members array can have duplicates, so this just returns an array of unique users.
```

Comment:
I've got this commented out for now while I think about a solution. Ultimately, I think this is a good check to have, it essentially ensures that the number of seats an account has in our DB is always inline with what Stripe has. However, it's causing an issue when someone initially starts a subscription. Currently, when someone starts a subscription, we start them at `seatsInFreeTier` + 1 number of seats (So when they put their credit card in, they're not charged $0.00). However, this causes the `subscriptionUpdated` webhook to get hit, and then this check sees there is a difference in the number of active seats in GB and in Stripe and updates it, causing a weird prorated refund to happen.